lib/msf/core/post/windows/services.rb

```diff
@@ -524,7 +524,7 @@ def service_restart(name, start_type=START_TYPE_AUTO, server=nil)
         vprint_good("[#{name}] Service started")
         return true
       else
-        raise status
+        raise status.to_s
```

Comment:
Not a bad idea however we should probably also update the documentation a bit before this to reflect this change. So should be like so:

```
#
  # Performs an aggressive service (re)start
  # If service is disabled it will re-enable
  # If service is running it will stop and restart
  #
  # @param name [String] The service name
  # @param start_type [Integer] The start type to configure if disabled
  # @param server [String] The server to target
  #
  # @return [Boolean] indicating success
  #
  #@raise [String] if OpenServiceA failed
  #
```

Reply:
Are you sure updating this is necessary ? Since, the raise is catch one line below. https://github.com/rapid7/metasploit-framework/blob/be48b1481ad80f79862c91a56bf2c584845b163e/lib/msf/core/post/windows/services.rb#L517-L536
I thought the raise will not exit the `service_restart` function, but maybe I'm wrong...

Reply:
Upon further inspection and talking with @smcintyre-r7 it looks like the code for this function may need reworking. Give us a bit and I'll try push up a proper fix for this to avoid the currently janky way of converting between numbers and strings that I think is part of the root cause of this issue.